### PR TITLE
feat:HP Compare two studies

### PR DESCRIPTION
**Pull request by wayfarer3130**
A hanging protocol to compare two series in different studies having the same series description.

### CODE REVIEW

File: ohif-configurable/hp-extension/src/hp/hpMN.ts [hunk 1]
```diff
@@ -37,19 +37,6 @@ export default [
             columns: 2,
           },
         },
-        displaySets: [
```
Comment (wayfarer3130) [inline ohif-configurable/hp-extension/src/hp/hpMN.ts:40]: This is just cleanup to agree with updated hanging protocol definition, does not affect the HP

File: ohif-configurable/hp-extension/src/hp/index.ts [hunk 2]
```diff
@@ -2,40 +2,40 @@ import hpHeart from './hpHeart';
...
 import hpBreast from './hpBreast';
 import hpVolume from './hpVolume';
+import hpCompare from './hpCompare';
 import ConfigPoint from "config-point";
 
 console.log("hpVolume=", hpVolume);
 
 const hangingProtocols = ConfigPoint.createConfiguration("hpExtensionHangingProtocols",
   {
-    // Include both protocols list and new list
```
Comment (wayfarer3130) [inline ohif-configurable/hp-extension/src/hp/index.ts:11]: Also cleanup

**Comment by wayfarer3130** [comment]
@ladeirarodolfo - could you review the change here please?

File: ohif-configurable/hp-extension/src/custom-attribute/sameAs.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/**
+ * This function extracts an attribute from the already matched display sets, and
+ * compares it to the attribute in the current display set, and indicates if they match.
+ * From 'this', it uses:
+ *    `sameAttribute` as the attribute name to look for
+ *    `sameDisplaySetId` as the display set id to look for
+ * From `options`, it looks for
```
Comment (ladeirarodolfo) [inline ohif-configurable/hp-extension/src/custom-attribute/sameAs.ts:7]: missing end of comment

File: ohif-configurable/hp-extension/src/custom-attribute/sameAs.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    return false;
+  }
+  const { displaySetInstanceUID } = match;
+  const altDisplaySet = displaySets.find(it => it.displaySetInstanceUID==displaySetInstanceUID);
+  if( !altDisplaySet ) {
+    console.log("No display set found with", displaySetInstanceUID, "in", displaySets);
+    return false;
+  }
+  const testValue = altDisplaySet[sameAttribute];
+  return testValue===displaySet[sameAttribute];
```
Comment (ladeirarodolfo) [inline ohif-configurable/hp-extension/src/custom-attribute/sameAs.ts:32]: lint comment: please, add spaces for === operator

File: ohif-configurable/hp-extension/src/custom-attribute/sameAs.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ * From 'this', it uses:
+ *    `sameAttribute` as the attribute name to look for
+ *    `sameDisplaySetId` as the display set id to look for
+ * From `options`, it looks for 
+ */
+export default function (displaySet, options) {
+  const { sameAttribute, sameDisplaySetId } = this;
+  if( !sameAttribute ) {
+    console.log("sameAttribute not defined in", this);
+    return `sameAttribute not defined in ${this.id}`;
```
Comment (ladeirarodolfo) [inline ohif-configurable/hp-extension/src/custom-attribute/sameAs.ts:13]: just for curiosity, is there any difference if I return a string or boolean?
It seems this message is being used only for logging purpose

File: ohif-configurable/hp-extension/src/hp/hpCompare.ts [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    name: 'Prior Compare',
+    hasUpdatedPriorsInformation: true,
+    numberOfPriorsReferenced: 1,
+    protocolMatchingRules: [
+      {
+        id: 'Two Studies',
+        weight: 1000,
+        attribute: 'StudyInstanceUID',
+        // The 'from' attribute says where to get the 'attribute' value from.  In this case
+        // prior means the second study in the study list.
```
Comment (ladeirarodolfo) [inline ohif-configurable/hp-extension/src/hp/hpCompare.ts:16]: Does this rule accept a prior being two subsequent study items but not indexes 0,1?

**Review by ladeirarodolfo — APPROVED** [review]
LGTM I have left only minor comments.

Thanks for your PR and btw thanks for the whole HP stuff effort, its very cool seeing something in a such complexity growing as it has being

File: ohif-configurable/hp-extension/package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@radicalimaging/hp-extension",
-  "version": "3.9.1",
+  "version": "3.9.3",
```
Comment (ladeirarodolfo) [inline ohif-configurable/hp-extension/package.json:3]: there is a released version 3.9.4, So the next value of this should be at least 3.9.5.
The value here will depend if there is CI routine which updates it at end

The same applies to config-mode package

**Pull request merged**